Formula/go.rb

```diff
@@ -32,22 +32,28 @@ class Go < Formula
       "linux-amd64"  => "4cdd2bc664724dc7db94ad51b503512c5ae7220951cac568120f64f8e94399fc",
     }
 
-    arch = "arm64"
-    platform = "darwin"
-
-    on_intel do
-      arch = "amd64"
+    version "1.17.13"
+
+    on_arm do
+      on_macos do
+        url "https://storage.googleapis.com/golang/go1.17.13.darwin-arm64.tar.gz"
```

Comment:
You may as well interpolate the version if you're going to define it above. It might need a separate variable though.